### PR TITLE
feat: add complete interpreter test coverage for DaemonSet

### LINKED ISSUE #6952: [Umbrella] Enhance third-party resource interpreter tests with output verification
/kind cleanup

We wish to enhance the testing framework for third-party resource interpreter customizations by adding support for verifying Lua script execution results against expected outputs.

In addition, the current test coverage for Lua script content is insufficient. Each operation of each resource has only one test case, which is not enough to fully cover the logic. Therefore, we also need to supplement the test cases for individual operations of individual resources.

Below, let's list all the subtasks that need to be completed for the current work item.

- [x] enhance resource interpreter test framework @zhzhuang-zju #6938 
- [ ] Complete test cases to cover the Lua script logic
  - [ ] apps.kruise.io
    - [ ] v1alpha1
      - [ ] AdvancedCronJob @XiShanYongYe-Chang #6991 
      - [ ] BroadcastJob
      - [ ] CloneSet
      - [ ] DaemonSet
      - [ ] SidecarSet
      - [ ] UnitedDeployment
    - [ ] v1beta1 - StatefulSet
  - [ ] argoproj.io - v1alpha1 - Workflow
  - [ ] flink.apache.org - v1beta1 - FlinkDeployment
  - [ ] helm.toolkit.fluxcd.io - v2beta1 - HelmRelease
  - [ ] kubeflow.org - v1 - TFJob
  - [ ] kustomize.toolkit.fluxcd.io - v1 - Kustomization
  - [ ] kyverno.io - v1
    - [ ] ClusterPolicy
    - [ ] Policy
  - [ ] source.toolkit.fluxcd.io
    - [ ] v1 - GitRepository
    - [ ] v1beta2
      - [ ] Bucket
      - [ ] HelmChart
      - [ ] HelmRepository
      - [ ] OCIRepository

If anyone is willing to get involved and complete some of these subtasks, we would be very grateful and welcome your participation. 

You can directly tell me in the comments which subtasks you want to complete, and I will assign them to you.

### PR DESCRIPTION
**Pull request by 7h3-3mp7y-m4n**
**What type of PR is this?**

#### Adds output-verification test cases for DaemonSet under third-party resource customizations, covering
- InterpretHealth

**What this PR does / why we need it**:
##### Improves test coverage for Kruise DaemonSet interpreter logic
**Which issue(s) this PR fixes**:

Fixes #6952 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by 7h3-3mp7y-m4n** [comment]
I'll fix the errors

**Comment by 7h3-3mp7y-m4n** [comment]
@XiShanYongYe-Chang and @mszacillo  can you look onto this ? I hope I got them right

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/assign

**Comment by XiShanYongYe-Chang** [comment]
Hi @FAUST-BENCHOU @Abhay349 would you like to help review this pr?

**Comment by Ferryx349** [comment]
@XiShanYongYe-Chang Sure, let me take a look..

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 1]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+      numberMisscheduled: 0
+      desiredNumberScheduled: 2
+      numberReady: 2
+      updatedNumberScheduled: 2
+      numberAvailable: 2
+      numberUnavailable: 0
+      daemonSetHash: 7dd59cf749
+
+---
+# case 2
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: you can remove case declaration here as it is already defined above.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 2]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+        daemonSetHash: hash-a
+        desiredNumberScheduled: 2
+        generation: 1
+        numberAvailable: 2
+        numberMisscheduled: 0
+        numberReady: 2
+        numberUnavailable: 0
+        observedGeneration: 1
+        resourceTemplateGeneration: 1
+        updatedNumberScheduled: 2
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: Add one extra blank line at the end of each test file.

**Comment by FAUST-BENCHOU** [comment]
sure
/assign

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 3]
```diff
@@ -1,5 +1,7 @@
 # test case for aggregating status of DaemonSet
-# case1. DaemonSet with two status items
+# case 1: DaemonSet with two status items
+# case 2: DaemonSet with no member status
+# case 3: DaemonSet with single member
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: You can consider adding some edge cases like stale member checking, picking last-empty dameonsetHash.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 4]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+      currentNumberScheduled: 0
+      numberMisscheduled: 0
+      desiredNumberScheduled: 0
+      numberReady: 0
+      updatedNumberScheduled: 0
+      numberAvailable: 0
+      numberUnavailable: 0
+      daemonSetHash: 0
+---
+
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: you can remove this this empty line.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 5]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+description: "Test AggregateStatus aggregates a single member DaemonSet"
+desiredObj:
+  apiVersion: apps.kruise.io/v1alpha1
+  kind: DaemonSet
+  metadata:
+    name: sample-single
+    namespace: test-kruise-daemonset
+    generation: 1
+operation: AggregateStatus
+statusItems:
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: Here u duplicated statusItems and operation, there should be only one statusItems block and should be before operation..

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 6]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+      numberReady: 2
+      numberAvailable: 2
+      updatedNumberScheduled: 2
+      numberMisscheduled: 0
+      numberUnavailable: 0
+      generation: 1
+      observedGeneration: 1
+      resourceTemplateGeneration: 1
+      daemonSetHash: hash-a
+output:
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: According to structure the output of this case is wrong, add proper spacing.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml [hunk 7]
```diff
@@ -70,3 +72,162 @@ statusItems:
...
+aggregatedStatus:
+apiVersion: apps.kruise.io/v1alpha1
+kind: DaemonSet
+metadata:
+    generation: 1
+    name: sample-single
+    namespace: test-kruise-daemonset
+operation: AggregateStatus
+status:
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: It seems the expected values are wrong, your expecteed output sets everything to 0, which contradicts logic. @XiShanYongYe-Chang @FAUST-BENCHOU Am I right?

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml [hunk 8]
```diff
@@ -1,5 +1,7 @@
 # test case for interpreting dependency of DaemonSet
 # case1. DaemonSet with configmap dependency
+# case2. DaemonSet with secret dependency
+# case3. DaemonSet with multiple containers sharing dependency
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml:6]: you can consider adding no dependency case as it is supported by logic.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml [hunk 9]
```diff
@@ -1,5 +1,8 @@
 # test case for interpreting health of DaemonSet
 # case1. DaemonSet interpreting health test
+# case2: Generation mismatch
+# case3: Not all updated pods are available
+# case4: Not all pods updated
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml:7]: you can consider adding the case of zero desired path as an edge case.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml [hunk 10]
```diff
@@ -52,3 +55,100 @@ observedObj:
...
+        labels:
+          app: sample-daemonset
+      spec:
+        containers:
+          - name: nginx
+            image: nginx:alpine
+  status:
+    observedGeneration: 2
+    desiredNumberScheduled: 5
+    updatedNumberScheduled: 3
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml]: You can run this command "git diff --check" to find any extra whitespace and fix them.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml [hunk 11]
```diff
@@ -52,3 +55,100 @@ observedObj:
...
+          - name: nginx
+            image: nginx:alpine
+  status:
+    observedGeneration: 2
+    desiredNumberScheduled: 5
+    updatedNumberScheduled: 3         
+    numberAvailable: 3
+operation: InterpretHealth
+output:
+  healthy: false
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interprethealth-test.yaml]: ditto

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml [hunk 12]
```diff
@@ -50,5 +52,96 @@ observedObj:
...
+          app: sample-daemonset
+      spec:
+        containers:
+          - name: nginx
+            image: nginx:alpine
+operation: InterpretStatus
+output:
+  status: {}
+---
+# Case 3:
```
Comment (Ferryx349) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml]: You can either fix description of this case to match with logic or move this case to aggregatestatus cuz mixed generation semantics are handled in aggregatestatus not in interpretstatus.

**Review by Abhay349 — CHANGES_REQUESTED** [review]
Most of the part seems good, there are issues in logic and structure at some points for which I have suggested changes and there are also some missing edge cases in some files.

**Comment by Ferryx349** [comment]
@XiShanYongYe-Chang @FAUST-BENCHOU
This is my very first PR review, so please feel free to point out anything I might have misunderstood or done incorrectly. I’d really appreciate your feedback and guidance, and apologies in advance if I’ve missed something.

Thanks :)

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 5 above)
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: Actually, we do not strictly require the order of each item, but putting output as the last item indeed helps with the review. 

This case does indeed have the statusItems and operation modules repeated.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 6 above)
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: You're right; there's a lack of indentation in the content below.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 7 above)
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: It seems so.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml (hunk 8 above)
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml:6]: What you pointed out is correct; you could improve the tone of the comment by being a bit gentler, which would make it even better.

Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml:6]: Thank you for pointing that out! I agree, and I’ll be more mindful about keeping the tone gentler in future comments.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 7 above)
Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: ya this part is strange for me.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 3 above)
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: considering add these two key logic
```Check if the member's status is updated to the latest generation```and
```Update the observed generation based on the observedResourceTemplateGenerationCount```

https://github.com/karmada-io/karmada/blob/d9abe6907035717bf1653fcbdb96ca2be171aecc/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/customizations.yaml#L78-L101

Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: ya just add about dameonsetHash part

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml [hunk 13]
```diff
@@ -1,5 +1,7 @@
 # test case for interpreting status of DaemonSet
 # case1. DaemonSet: interpret status test
+# case2. DaemonSet: empty observed status
+# case3. DaemonSet: multiple members but not all observed generation updated
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml:8]: considering add test about function `tonumber`
https://github.com/karmada-io/karmada/blob/d9abe6907035717bf1653fcbdb96ca2be171aecc/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/customizations.yaml#L140

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml (hunk 12 above)
Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml]: seems no mixed generation members logic in statusReflection

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 3 above)
Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: your mixed observedGeneration logic is better to be here

Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: also consider to add cases about mixed nil status:If some fields in the status of certain members are nil, the accumulation should be skipped.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 1 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: well i tried to keep the same as mentioned in other v1alpha1 `testdata` example [here](https://github.com/karmada-io/karmada/blob/master/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/BroadcastJob/testdata/aggregatestatus-test.yaml#L77)
but as you say I'll make the changes

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 4 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: yeah I'll lint it

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 2 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: for sure!

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml (hunk 8 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretdependency-test.yaml:6]: yeah I saw on the lua logic , and i almost tried to add it but i encountered some error so i was hoping to get it fix and push the changes ,  But thanks for pointing it out :)

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml (hunk 13 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml:8]: for sure!

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml (hunk 12 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/interpretstatus-test.yaml]: My bad I'll make changes ASAP!

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 3 above)
Comment (7h3-3mp7y-m4n) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: for sure I'll add them

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml (hunk 1 above)
Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/testdata/aggregatestatus-test.yaml]: Totally agree with your point. I actually used this declaration initially 😅, but later realized that declaring it here doesn’t add much value.

**Comment by XiShanYongYe-Chang** [comment]
Hi @7h3-3mp7y-m4n have the existing review comments been fixed?

**Comment by XiShanYongYe-Chang** [comment]
hi @7h3-3mp7y-m4n, are you still working on this?

**Comment by FAUST-BENCHOU** [comment]
@7h3-3mp7y-m4n If you're too busy to work on this PR right now, I can help you improve it based on our discussions within the next couple of days.

**Comment by Ferryx349** [comment]
@FAUST-BENCHOU I agree with you and also happy to help wherever it makes sense, just let me know.
Thanks!

**Comment by XiShanYongYe-Chang** [comment]
Let's wait another week and see. Maybe @7h3-3mp7y-m4n is busy with other things. If there is still no response, we can continue to make modifications based on his commit.

**Comment by 7h3-3mp7y-m4n** [comment]
Hey folks,

Apologies for the delay. All of the work has been completed, except for the part I opted to work on, which has been pending for over a month now, I’m really sorry about that.

Over the past few weeks, I ran into some medical issues, along with increased work pressure. Thankfully, I’m doing well now and back on track. I’ve almost finished the PR. I added more tests based on the suggestions provided and also included a line of Lua code for `dependencyInterpretation`, which allows us to write a test for a DaemonSet with no dependencies.

The only remaining part is the `aggregateStatus` section, where I’m a bit confused. Based on the Lua logic, whenever I try to implement it, the output always comes out as 0. I also noticed that we’re using the DaemonSet hash as a numeric 0. Shouldn’t this be a string ("") instead?

I’m not very experienced with Lua scripting, so I might be missing something here. I’d really appreciate your help in completing the last few tests related to `aggregateStatus`.

Once again, I apologize for the delay, and thank you for your patience. @XiShanYongYe-Chang @Abhay349 @FAUST-BENCHOU

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/customizations.yaml [hunk 14]
```diff
@@ -162,5 +162,8 @@ spec:
         local kube = require("kube")
         function GetDependencies(desiredObj)
           refs = kube.getPodDependencies(desiredObj.spec.template, desiredObj.metadata.namespace)
+          if refs == nil then
+            refs = {}
+          end
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1alpha1/DaemonSet/customizations.yaml:167]: i dont think we need to change this......

**Comment by FAUST-BENCHOU** [comment]
@7h3-3mp7y-m4n no worries and thanks for your hardwork! I've added some changes you needed and submitted a PR application to merge it into your current PR process. @Abhay349 could help me to review my pr in https://github.com/7h3-3mp7y-m4n/karmada/pull/1 ? Since i only add tests based on my previous comment......
BTW I'm not sure @7h3-3mp7y-m4n  if you can merge my branch directly, because merging commits is not allowed.Maybe u can just consider this branch as a modify advice....

**Comment by XiShanYongYe-Chang** [comment]
@7h3-3mp7y-m4n It's okay, no need to apologize. I hope you get well soon and have a good rest.

**Comment by XiShanYongYe-Chang** [comment]
Thanks a lot @FAUST-BENCHOU 

@7h3-3mp7y-m4n After the review is complete, you can merge the commit provided by @ using the command. This will require the use of the `git cherry-pick` command.

Before that, could you help compress your current commits into one?

**Comment by Ferryx349** [comment]
@7h3-3mp7y-m4n no worries at all — hope you recover soon!

**Comment by 7h3-3mp7y-m4n** [comment]
Thanks All :)
For sure I'll squash my commits @XiShanYongYe-Chang

**Comment by 7h3-3mp7y-m4n** [comment]
Crap i messed the rebase. I'll take the inspiration from @FAUST-BENCHOU  and raise a new pr

**Comment by XiShanYongYe-Chang** [comment]
> Crap i messed the rebase. I'll take the inspiration from @FAUST-BENCHOU and raise a new pr

Hi @7h3-3mp7y-m4n This is a good opportunity to familiarize yourself with Git operations.

**Comment by 7h3-3mp7y-m4n** [comment]
You are right @XiShanYongYe-Chang , I often find it difficult while cherry pick,  for sure I'll look  into this
